Source/WebKit/UIProcess/ProvisionalFrameProxy.cpp

```diff
@@ -39,22 +39,16 @@ namespace WebKit {
 
 WTF_MAKE_TZONE_ALLOCATED_IMPL(ProvisionalFrameProxy);
 
-ProvisionalFrameProxy::ProvisionalFrameProxy(WebFrameProxy& frame, Ref<FrameProcess>&& frameProcess)
+ProvisionalFrameProxy::ProvisionalFrameProxy(WebFrameProxy& frame, Ref<FrameProcess>&& frameProcess, CommitTiming commitTiming)
     : m_frame(frame)
     , m_frameProcess(WTFMove(frameProcess))
     , m_visitedLinkStore(frame.page()->visitedLinkStore())
 {
     Ref process = this->process();
     process->markProcessAsRecentlyUsed();
-    process->send(Messages::WebFrame::CreateProvisionalFrame(ProvisionalFrameCreationParameters {
-        frame.frameID(),
-        std::nullopt,
-        frame.layerHostingContextIdentifier(),
-        frame.effectiveSandboxFlags(),
-        frame.effectiveReferrerPolicy(),
-        frame.scrollingMode(),
-        frame.remoteFrameSize()
-    }), frame.frameID());
+    auto parameters = frame.defaultProvisionalFrameCreationParameters();
+    parameters.commitTiming = commitTiming;
```

Comment:
nit: I think it would be a slightly nicer design to pass commitTiming to the call to defaultProvisionalFrameCreationParameters as a parameter.  Also, I'd rename defaultProvisionalFrameCreationParameters to provisionalFrameCreationParameters